packages/good-design/src/core/buttons/ClaimButton.tsx

```diff
@@ -60,69 +114,78 @@ const ClaimButton = ({
                   Action Required
                 </Title>
                 <Text color={textColor} fontFamily="subheading" fontWeight="normal" fontSize="md">
-                  To complete this action, sign in your wallet.
+                  To complete this action, continue in your wallet.
                 </Text>
               </>
             ),
-            body: (
-              <Box mb="24px" mt="16px">
-                <Title fontFamily="subheading" fontWeight="normal" fontSize="sm">
-                  WATCH
-                </Title>
-                <Text fontFamily="mono" color="primary" mb="8px" fontSize="l" fontWeight="bold">
-                  What is signing?
-                </Text>
-                <Image source={ClaimImage} w="full" h="auto" />
-              </Box>
-            ),
+            body: loading || claimLoading ? <Spinner color="emerald.500" /> : actionModalBody.verify.body,
+            footer: isWhitelisted || loading || claimLoading ? undefined : actionModalBody.verify.footer,
             closeText: "",
             hasBottomBorder: false
           },
-    [firstClaim, textColor]
+    [firstClaim, textColor, loading, isWhitelisted, claimLoading]
   );
 
-  const handleClaimCall = useCallback(
+  const handleClaim = async (first: boolean) => {
+    try {
+      const success = await claim();
+
+      if (success !== true || first === false) {
+        return;
+      }
+
+      showFirstClaimModal();
+    } finally {
+      setClaimLoading(false);
+      hideActionModal();
+    }
+  };
+
+  const handleModalOpen = useCallback(
     async (first = false) => {
       setFirstClaim(first);
       showActionModal();
-
-      try {
-        const success = await claim();
-        if (success !== true || first === false) return;
-
-        showFirstClaimModal();
-      } finally {
-        hideActionModal();
+      if (isWhitelisted) {
+        setClaimLoading(true);
+        await handleClaim(first);
+      } else if (fuseWhitelisted && syncStatus) {
+        const success = await syncStatus;
+        if (success) {
+          setClaimLoading(true);
+          await handleClaim(true);
+        } else {
+          //// what to do here? tjis should not continue with FV Flow
+        }
+      } else {
+        // here nothing happens, FV flow starts
+        // what to do with an edge case where fuseWhitelisted or syncStatus might not be set or
+        // give a value in time before user interaction is done
       }
```

Comment:
just simplify this by now
```suggestion
      if (isWhitelisted) {
        setClaimLoading(true);
        await handleClaim(first);
        return;
      } 
      
      if (fuseWhitelisted && syncStatus) {
        const success = await syncStatus;
        
        if (!success) {
          return;
        }
        
        setClaimLoading(true);
        await handleClaim(true);
      }          
```

Reply:
removed for now